app/controllers/users_controller.rb

```diff
@@ -0,0 +1,14 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def posts
+    @user = User.find(params[:id])
+    @posts = @user.posts.order(created_at: :desc)
+  end
```

Comment:
:red_circle: [REQUIRED] I noticed that you have a `posts` method that returns a list :books:  of posts for a given user :bust_in_silhouette: . This method is currently defined in the `UsersController` class, but it would be better to define it in the `PostController` class as the index method.

Here are a few reasons why this is better:

1. It follows the RESTful convention of having a resource controller handle all requests for that resource. In this case, the resource is posts, so the PostController should be the one handling all requests for posts.

2. It makes the code more modular and easier to understand. The UsersController class should only be responsible for handling requests for users, and the PostController class should only be responsible for handling requests for posts.

I recommend that you make the following changes to your code:

- Move the posts method to the PostController class.
- Change the method name from posts to the index to follow the RESTful convention.
- Update the route for the posts action to point to the PostController class.
- Move `posts.html.erb` from` views/users `to` views/posts`
- Finally, Rename `posts.html.erb` to` index.html.erb`